Make checkout button sticky footer on mobile

haszari opened the pull request:
Fixes #1873

This PR adds a class to the the CTA / checkout button component, and adds styling so that displays as a fixed footer with shadow on smaller screens.

There's a mobile footer in storefront, which has icon links to account / search / cart. This PR also includes a style tweak to hide this footer on the cart page (on mobile only), so it doesn't interfere with the cart button. It's not needed and is potentially distracting to show these other links on the cart page.

### Screenshots

Here's a screenshot, with theme _Galleria_ (child theme of _Storefront_):

<img width="376" alt="Screen Shot 2020-03-10 at 4 22 25 PM" src="https://user-images.githubusercontent.com/4167300/76276688-66d08680-62eb-11ea-989a-ec8bb8f5cbe0.png">

Here's a screenshot on iPhone with [_Neve_](https://wordpress.org/themes/neve/) from WPORG themes:

![IMG_E17A4D61B5B9-1](https://user-images.githubusercontent.com/4167300/76276793-b6af4d80-62eb-11ea-9774-25b6a61338ed.jpeg)

(For fun I tested this with a few different themes.)

### How to test the changes in this Pull Request:

1. Add the cart block to a page, publish.
2. Set that page as the cart page in `WooCommerce > Settings > Advanced`. 
3. Add some items to cart.
6. Test the cart page on various devices / browsers / screen sizes. The cart should work correctly, and on smaller screens, the checkout button should be prominent and easy to find :)

Aljullu commented on assets/js/blocks/cart-checkout/cart/full-cart/style.scss:
```diff
@@ -404,6 +404,23 @@ table.wc-block-cart-items {
+	// Note - this will only work on woo "Cart" page.
+	.woocommerce-page.woocommerce-cart.theme-storefront .storefront-handheld-footer-bar {
+		display: none;
+	}
```
I think it's a great idea hiding this bar in the _Cart_ page, I would even hide it in _Checkout_ page as well, how does it sound?

My only concern is that I wonder if this code belongs to the Storefront theme instead of here. Stores using WooCommerce with other themes which are not Storefront don't need to load this CSS. I also think it would be easy to maintain it there than in this repo.

nerrad replied:
Agreed with this. I think we should avoid theme specific styling in the plugin unless absolutely necessary.

haszari replied:
I'll revert this and log a follow up issue. It is essential that we fix this for Storefront, which we could potentially do in storefront directly or in a "theme support" stylesheet (similar to woo core approach).

<img width="459" alt="Screen Shot 2020-03-11 at 9 45 33 AM" src="https://user-images.githubusercontent.com/4167300/76357728-173a9b00-637d-11ea-8d3b-83acfd3ea8dc.png">

Logged here: #1930

haszari commented:
Thanks for the reviews @Aljullu & @nerrad. Ready for another review :)

I've removed the storefront-specific fix (and logged a follow up #1930). Here are some screenshots to show this in other themes - Twenty Sixteen and Twenty Twenty.

<img width="392" alt="Screen Shot 2020-03-11 at 10 00 15 AM" src="https://user-images.githubusercontent.com/4167300/76360713-6931ef80-6382-11ea-8d1a-9c35238b88d5.png">
<img width="366" alt="Screen Shot 2020-03-11 at 10 22 44 AM" src="https://user-images.githubusercontent.com/4167300/76360716-6c2ce000-6382-11ea-8e24-3c23e0c41328.png">

> These changes will make it not possible to use the Cart block in a page that is not set as the Cart page, is that ok

This is now reverted (see screenshot above - cart on homepage in twenty twenty).
